lua/darkrp_modules/levels/sv_addways.lua

```diff
@@ -1,62 +1,106 @@
+				local victime = victim:Nick()
+				local money = DarkRP.formatMoney(LevelSystemConfiguration.TakeAwayMoneyAmount)
+				local XP = killer:addXP( 10 * (victim:getDarkRPVar('level') or 1), true)
+
+				if (LevelSystemConfiguration.Friendly) then
+					if ((killer:getDarkRPVar('level') or 1) <= (victim:getDarkRPVar('level') or 1)) then
 						killer:addMoney(LevelSystemConfiguration.TakeAwayMoneyAmount)
-						DarkRP.notify(killer, 0,4,LevelSystemConfiguration.LangKillNotify3)
+						DarkRP.notify(killer, 0,4, string.format( LevelSystemConfiguration.LangKillNotify, XP, money, victime ))
+						if guthlogsystem then
+							guthlogsystem.addLog( "DarkRP Leveling System", "*"..killer.."* got &"..XP.."& XP and &"..money.."& for killing ?"..victime.."?" )
+						end
+
+						if (victim:canAfford(LevelSystemConfiguration.TakeAwayMoneyAmount)) then
+							victim:addMoney(-LevelSystemConfiguration.TakeAwayMoneyAmount)
+							DarkRP.notify(victim, 0,4, string.format( LevelSystemConfiguration.LangDeath, money ))
+						end
+					else 
+						DarkRP.notify(killer,0,4, string.format( LevelSystemConfiguration.LangKillNotify2, XP, victime ))
+						if guthlogsystem then
+							guthlogsystem.addLog( "DarkRP Leveling System", "*"..killer:Name().."* got &"..XP.."& XP for killing ?"..victime.."?" )
+						end
 					end
-						
+				else
+					killer:addMoney(LevelSystemConfiguration.TakeAwayMoneyAmount)
+					DarkRP.notify(killer, 0,4, string.format( LevelSystemConfiguration.LangKillNotify3, victime ))
+                    if guthlogsystem then
+						guthlogsystem.addLog( "DarkRP Leveling System", "*"..killer:Name().."* got &0& XP ?"..victime.."? (Friendly mode no XP gain)")
+                    end
 				end
+			end
 		end
 	end
 end
 
-hook.Add( "PlayerDeath", "manolis:MVLevels:PlayerDeathBC", PlayerDeath )
+hook.Add( 'PlayerDeath', 'manolis:MVLevels:PlayerDeathBC', PlayerDeath )
 
 
-function NPCDeath(npc, killer,weapon)
-	if(LevelSystemConfiguration.NPCXP) then
-		if(npc != killer) then // Not a suicide? Somehow.
-			if(killer:IsPlayer()) then
-				local XP = killer:addXP(npc:GetVar("GiveXP") or LevelSystemConfiguration.NPCXPAmount, true)
-				if(XP) then
-					DarkRP.notify(killer, 0,4,LevelSystemConfiguration.LangKillNPC)
+local function NPCDeath(npc, killer,weapon)
+	if (LevelSystemConfiguration.NPCXP) then
+		if (npc != killer) then -- Not a suicide? Somehow.
+			if (killer:IsPlayer()) then
+				local XP = killer:addXP(npc:GetVar('GiveXP') or LevelSystemConfiguration.NPCXPAmount, true)
+				if (XP) then
+					DarkRP.notify(killer, 0,4, string.format( LevelSystemConfiguration.LangKillNPC, XP ))
+                    if guthlogsystem then
+						guthlogsystem.addLog( "DarkRP Leveling System", "*"..killer:Name().."* got &"..XP.."& XP for killing an NPC" )
+                    end
 				end
 			end
 		end
 	end
 end
 
-hook.Add( "OnNPCKilled", "manolis:MVLevels:OnNPCKilledBC", NPCDeath )
+hook.Add( 'OnNPCKilled', 'manolis:MVLevels:OnNPCKilledBC', NPCDeath )
```

Comment:
You don't have to change this but if you're going to format the code like
hook.Add( "OnNPCKilled", ... NPCDeath ) 
then make sure you do it everywhere so it's consistent